tests/process/spawn/stream.luau

```diff
@@ -0,0 +1,21 @@
+local process = require("@lune/process")
+
+-- Should be able to write and read from child process streams
+
+local msg = "hello, world"
+
+local catChild = process.spawn("cat")
+catChild.stdin:write(msg)
+assert(
+	msg == buffer.tostring(catChild.stdout:read(#msg)),
+	"Failed to write to stdin or read from stdout of child process"
+)
+
+local echoChild = if process.os == "windows"
+	then process.spawn("echo", { msg, "1>&2" }, { shell = "cmd" })
```

Comment:
Try carrying out the command using `/c` and anything after would be run as command then terminate.

In windows, `cmd` alone would spawn as an interactive terminal instead running your command directly.
```lua
then process.spawn("/c", { "echo", msg, "1>&2" }, { shell = "cmd" })
```
Similar to `cmd /c echo {msg} 1>&2`

Reply:
Don't know if providing "/c" with shell set would work, probably need to invoke it as an exe.

Reply:
Should probably make this an issue (maybe?) since if I set the shell key, I expect it to run within my shell (just like setting sh should try `-c`).

Reply:
The method I provided worked on my windows 11 machine. And as far as I know `cmd` does not support `-c`, only `/c` so yeah that would somewhat of an OS compatibility issue in `lune`,